src/sentry/uptime/subscriptions/subscriptions.py

```diff
@@ -211,6 +218,24 @@ def create_project_uptime_subscription(
         uptime_status=uptime_status,
     )
 
+    data_source = DataSource.objects.create(
+        type=DATA_SOURCE_UPTIME_SUBSCRIPTION,
+        organization=project.organization,
+        source_id=str(uptime_subscription.id),
+    )
+    detector = Detector.objects.create(
+        type=UptimeDomainCheckFailure.slug,
+        project=project,
+        name=name,
+        owner_user_id=owner_user_id,
+        owner_team_id=owner_team_id,
+        config={
+            "environment": environment.name if environment else None,
+            "mode": mode.value,
+        },
+    )
+    DataSourceDetector.objects.create(data_source=data_source, detector=detector)
```

Comment:
Let's wrap this all in a transaction. Actually, we should already have a transaction around  the original row creation, so we can wrap all of that as part of this.

Reply:
Where's the transaction around the original row creation?